components/registry-facade/pkg/blobserve/blobserve.go

```diff
@@ -126,18 +127,40 @@ func (reg *Server) serve(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	var fs http.FileSystem
-	fs = blob
-	if workdir != "" {
-		fs = prefixingFilesystem{Prefix: workdir, FS: fs}
-	}
-
 	log.WithField("path", req.URL.Path).Debug("handling blobserve")
 	pathPrefix := fmt.Sprintf("/%s:%s", repo, tag)
 	if req.URL.Path == pathPrefix {
 		req.URL.Path += "/"
 	}
 
+	// http.FileServer has a special case where ServeFile redirects any request where r.URL.Path
+	// ends in "/index.html" to the same path, without the final "index.html".
+	// We do not want this behaviour to make the gitpod-ide-index mechanism in ws-proxy work.
+	if strings.TrimPrefix(req.URL.Path, pathPrefix) == "/index.html" {
+		fn := filepath.Join(workdir, "index.html")
```

Comment:
Does `http.FileSystem` come with some form of caching build-in? It feels expensive to do the `open`/`stat` per request.

Reply:
If we'd `stat` for each and every request, I'd totally agree. Here we just do that for requests to `/index.html` which should not happen all that often.